src/cmd/keys.js

```diff
@@ -617,27 +588,54 @@ module.exports = class KeysCommand {
 			if (!device.isInDfuMode) {
 				device = await usbUtils.reopenInDfuMode(device);
 			}
-			this._setDfuId(device);
+			this.platform = device._info.type;
 			return device;
 		} catch (err) {
 			throw new VError(ensureError(err), 'Unable to get DFU device');
 		}
 	}
 
-	_validateSegmentSpecs(segmentName) {
-		let specs = deviceSpecs[this.dfuId] || {};
-		let params = specs[segmentName];
+	_validateSegments(segmentName) {
+		let segments = this._getAllSegments() || {};
```

Comment:
Given that you don't change `segments` and `params` after creation you can put `const` instead of `let`

Reply:
yes! surprised that lint didn't warn this time

Reply:
Regarding this https://github.com/particle-iot/particle-cli/pull/675/files#r1340443296

You can do:
```suggestion
		const segments = this._getAllSegments();
		const params = segments ? segments[segmentName] : null;
```

Reply:
Checking more deep the way getAllSegments works you always will receive an object. there is no way to get a null. So the `|| {}` can be removed